### ISSUE TITLE
Multiple commands for a keybinding

### ISSUE THREAD

**Issue by adam-mcdaniel** [kkawakam/rustyline#306]
It would be a really handy feature to add a `Cmd::Multiple(Vec<Cmd>)` enumeration for the `Cmd` enum so that multiple commands could be used in succession for a keybinding.

**Comment by gwenn** [comment]
WIP: #293

**Comment by nu8** [comment]
https://github.com/nushell/nushell/issues/2421

**Comment by 89z** [comment]
https://github.com/nushell/nushell/issues/2781

**Comment by 89z** [comment]
https://github.com/kkawakam/rustyline/pull/492

**Comment by 89z** [comment]
so should this be closed since #492 was merged?

**Comment by gwenn** [comment]
> so should this be closed since #492 was merged?

No, #492 makes possible to put condition(s) on which command is executed depending on input state.
But still only one command can be executed.

**Comment by superiums** [comment]
still looking for this feature. a common use case is: after replace string, we need move cursor.  is there any way to finish this ?

**Comment by hdwalters** [comment]
> I see that enum EventHandler has this TODO comment

With that in mind, I would not recommend adding a new `EventHandler::Macro` enum value; it seems much better to add a new `Cmd::Multiple(Vec<Cmd>)` as per the OP's request. Note, it would have to be `Cmd::Multiple(Vec<Box<Cmd>>)` because `Cmd` cannot be sized with this addition.
